iotdb/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -458,6 +468,22 @@ public QueryDataSource query(String deviceId, String measurementId, QueryContext
     }
   }
 
+  /**
+   * returns the top k% sensors most frequently used in queries.
+   */
+  public Set calTopKSensor(String sensorId, double k) {
+    int num = (int) (lruForSensorUsedInQuery.size() * k) + 1;
+    Set<String> sensorSet = new HashSet<>(num);
+
+    for (String sensor : lruForSensorUsedInQuery) {
+      if (--num > 0) {
+        sensorSet.add(sensor);
+      }
```

Comment:
I would suggest a "break" here when num <= 0, you do not have to iterate over the whole set.

Reply:
fixed